src/deviation.jl

```diff
@@ -75,11 +68,10 @@ Compute the L1 distance between two arrays: ``\\sum_{i=1}^n |a_i - b_i|``.
 Efficient equivalent of `sum(abs, a - b)`.
 """
 function L1dist(a::AbstractArray{T}, b::AbstractArray{T}) where T<:Number
-    n = length(a)
-    length(b) == n || throw(DimensionMismatch("Input dimension mismatch"))
+    length(a) == length(b) || throw(DimensionMismatch("Input dimension mismatch"))
     r = 0.0
```

Comment:
Shouldn't this check axes not length? Although `eachindex(a, b)` performs such a check, in which case probably better to delete it:
```
julia> eachindex(OffsetArray([1,2,3], 1), OffsetArray([1,2,3], 2))
ERROR: DimensionMismatch: all inputs to eachindex must have the same indices, got OffsetArrays.IdOffsetRange(values=2:4, indices=2:4) and OffsetArrays.IdOffsetRange(values=3:5, indices=3:5)
```
Also, why 0.0 not `zero(T)`? (And why restrict to matching eltypes, `T==T`?)

Reply:
> Shouldn't this check axes not length?

No, arrays don't have to have matching sizes, just the same length. Enforcing this would be breaking.

> Also, why 0.0 not zero(T)? (And why restrict to matching eltypes, T==T?)

That's ancient code. I think it would be good to keep the PR focused on indexing, then it's easier and probably faster to review, merge, and release.

Reply:
I guess I was going by this:
```
Efficient equivalent of `sum(abs, a - b)`
```
I now notice zip actually enforces this anyway:
```
julia> using OffsetArrays

julia> collect(zip(OffsetArray([1,2,3], 1), OffsetArray([1,2,3], 2)))
ERROR: DimensionMismatch: dimensions must match: a has dims (OffsetArrays.IdOffsetRange(values=2:4, indices=2:4),), b has dims (OffsetArrays.IdOffsetRange(values=3:5, indices=3:5),), mismatch at 1
Stacktrace:
```

Reply:
`zip` doesn't enforce any axes (not even the `length` and hence the check is necessary):
```julia
julia> using OffsetArrays

julia> foreach(println, zip(OffsetArray([1,2,3], 1), OffsetArray([1,2,3], 2)))
(1, 1)
(2, 2)
(3, 3)

julia> foreach(println, zip(OffsetArray([1 2 3], 1, 1), OffsetArray([1,2,3], 2)))
(1, 1)
(2, 2)
(3, 3)

julia> foreach(println, zip(OffsetArray([1,2,3], 1), OffsetArray([1,2,3,4], 2)))
(1, 1)
(2, 2)
(3, 3)

julia> foreach(println, zip(OffsetArray([1 2 3], 1, 1), OffsetArray([1,2,3,4], 2)))
(1, 1)
(2, 2)
(3, 3)
```

Reply:
Indeed, it seems like you can iterate it, even when `axes` would throw:
```
julia> axes(zip([1 2; 3 4], [1 2 3 4]))
ERROR: DimensionMismatch: dimensions must match: a has dims (Base.OneTo(2), Base.OneTo(2)), b has dims (Base.OneTo(1), Base.OneTo(4)), mismatch at 1
```

Reply:
Maybe this can be marked resolved.